pkg/sql/opt/optbuilder/testdata/join

```diff
@@ -2825,3 +2825,32 @@ project
       └── projections
            ├── COALESCE(abcd.d:4, dxby.d:8) [as=d:15]
            └── COALESCE(abcd.b:2, dxby.b:10) [as=b:16]
+
+# Regression for #79886
+exec-ddl
+CREATE TABLE tab2 (pk INT8 PRIMARY KEY, col0 INT8, col1 FLOAT8, col3 INT8)
```

Comment:
I think this test belongs in `pkg/sql/opt/xform/testdata/rules/join` instead of optbuilder tests. Maybe you can use a table that is already used in the `GenerateLookupJoins` tests since the reproduction seems to be pretty general.

Reply:
I moved it but didn't spend anytime trying to re-use tables.  Actually I think it would be cool if opttester supported a subtest for things like this for both CLI test running granularity and test isolation.  Whaddya think of that?